src/Compilers/CSharp/Portable/Symbols/ReferenceManager.cs

```diff
@@ -219,27 +219,49 @@ public void CreateSourceAssemblyForCompilation(CSharpCompilation compilation)
             /// Used by EnC to create symbols for emit baseline. The PE symbols are used by <see cref="CSharpSymbolMatcher"/>.
             /// 
             /// The assembly references listed in the metadata AssemblyRef table are matched to the resolved references 
-            /// stored on this <see cref="ReferenceManager"/>. Each AssemblyRef is matched against the assembly identities
-            /// using an exact equality comparison. No unification or further resolution is performed.
+            /// stored on this <see cref="ReferenceManager"/>. We assume that the dependencies of the baseline metadata are 
+            /// the same as the dependencies of the current compilation. This is not exactly true when the dependencies use 
+            /// time-based versioning pattern, e.g. AssemblyVersion("1.0.*"). In that case we assume only the version
+            /// changed and nothing else.
+            /// 
+            /// Each AssemblyRef is matched against the assembly identities using an exact equality comparison modulo version. 
+            /// AssemblyRef with lower version in metadata is matched to a PE assembly symbol with the higher version 
+            /// (provided that the assembly name, culture, PKT and flags are the same) if there is no symbol with the exactly matching version. 
+            /// If there are multiple symbols with higher versions selects the one with the minimal version among them.
+            /// 
+            /// Matching to a higher version is necessary to support EnC for projects whose P2P dependencies use time-based versioning pattern. 
+            /// The versions of the dependent projects seen from the IDE will be higher than 
+            /// the one written in the metadata at the time their respective baselines are built.
+            /// 
+            /// No other unification or further resolution is performed.
             /// </remarks>
-            public PEAssemblySymbol CreatePEAssemblyForAssemblyMetadata(AssemblyMetadata metadata, MetadataImportOptions importOptions)
+            public PEAssemblySymbol CreatePEAssemblyForAssemblyMetadata(AssemblyMetadata metadata, MetadataImportOptions importOptions, out ImmutableDictionary<AssemblyIdentity, AssemblyIdentity> assemblyReferenceIdentityMap)
```

Comment:
Consider moving/copying it to the doc-comment for the method so that its contract could be understood without reading its body.